docs/rules/no-unused-expressions.md

```diff
@@ -39,6 +39,21 @@ a && b()
 a, b()
 
 c = a, b;
+
+a() && function namedFunctionInExpressionContext () {f();}
+
+(function anIncompleteIIFE () {});
+
```

Comment:
Uh oh, did I click Comment yet, or not? If did already, sorry for the duplicate; if not yet, now is the time!

1. Looking ahead to editing which will happen within the next few weeks (more about this after these pull requests have been accepted :) it would help to move the subject of directives to the end of **Rule Details**, after the note about sequence expressions.

2. Suggest bringing together earlier proposed wording that this rule *does not apply* to directives with this nice new description of *where they occur*.

3. Recommend the 2 clear cut examples in separate “code wells.”

This rule does not apply to directives (which are in the form of literal string expressions at the beginning of a script, module, or function).

```js
"use strict";
```

```js
"use asm";
```